cmd/gh2gcs/cmd/root.go

```diff
@@ -0,0 +1,197 @@
+	PersistentPreRunE: initLogging,
+	RunE: func(*cobra.Command, []string) error {
+		return run(opts)
+	},
+}
+
+type options struct {
+	org                string
+	repo               string
+	tag                string
+	includePrereleases bool
+	bucket             string
+	releaseDir         string
+	outputDir          string
+	logLevel           string
+}
+
+var opts = &options{}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().StringVar(
+		&opts.org,
+		"org",
+		// TODO: Remove test value
+		"containernetworking",
+		// TODO: Improve usage text
+		"org",
+	)
+
+	rootCmd.PersistentFlags().StringVar(
+		&opts.repo,
+		"repo",
+		// TODO: Remove test value
+		"plugins",
+		// TODO: Improve usage text
+		"repo",
+	)
+
+	// TODO: This should be a string array to accept multiple tags
+	rootCmd.PersistentFlags().StringVar(
+		&opts.tag,
+		"tag",
+		"",
+		// TODO: Improve usage text
+		"tag",
+	)
+
+	rootCmd.PersistentFlags().BoolVar(
+		&opts.includePrereleases,
+		"include-prereleases",
+		false,
+		// TODO: Improve usage text
+		"include-prereleases",
+	)
+
+	rootCmd.PersistentFlags().StringVar(
+		&opts.bucket,
+		"bucket",
+		// TODO: Remove test value
+		"k8s-staging-release-test",
+		// TODO: Improve usage text
+		"bucket",
+	)
+
+	rootCmd.PersistentFlags().StringVar(
+		&opts.releaseDir,
+		"release-dir",
+		// TODO: Remove test value
+		"augustus/release",
+		// TODO: Improve usage text
+		"release-dir",
+	)
+
+	rootCmd.PersistentFlags().StringVar(
+		&opts.outputDir,
+		"output-dir",
+		"",
+		// TODO: Improve usage text
+		"output-dir",
+	)
+
+	rootCmd.PersistentFlags().StringVar(
+		&opts.logLevel,
+		"log-level",
+		"info",
+		"the logging verbosity, either 'panic', 'fatal', 'error', 'warn', 'warning', 'info', 'debug' or 'trace'",
+	)
+}
+
+func initLogging(*cobra.Command, []string) error {
+	return log.SetupGlobalLogger(opts.logLevel)
+}
+
+func run(opts *options) error {
+	if err := gcp.PreCheck(); err != nil {
+		return errors.Wrap(err, "pre-checking for GCP package usage")
+	}
+
+	if opts.outputDir == "" {
+		tmpDir, err := ioutil.TempDir("", "gh2gcs")
+		if err != nil {
+			return err
+		}
+
+		opts.outputDir = tmpDir
+	}
+
+	// Create a real GitHub API client
+	gh := github.New()
+
+	// TODO: Support downloading releases via yaml config
+	uploadConfig := &gh2gcs.Config{}
+	releaseConfig := &gh2gcs.ReleaseConfig{
+		Org:        opts.org,
+		Repo:       opts.repo,
+		Tags:       []string{},
+		GCSBucket:  opts.bucket,
+		ReleaseDir: opts.releaseDir,
+	}
+
+	if opts.tag != "" {
+		releaseConfig.Tags = append(releaseConfig.Tags, opts.tag)
+	} else {
+		releaseTags, err := gh.GetReleaseTags(opts.org, opts.repo, opts.includePrereleases)
+		if err != nil {
+			return err
```

Comment:
I suggest wrapping those errors for a better debug-ability:

```suggestion
			return errors.Wrap(err, "getting release tags")
```